flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -513,6 +519,56 @@ public static Configuration cloneConfiguration(Configuration configuration) {
 		return clonedConfiguration;
 	}
 
+	/**
+	 * Format the default gc logging options
+	 * @param logDirectory to save the gc log
+	 * @param conf flink configuration
+	 * @return the formatted gc logging options string
+	 */
+	public static String getGCLoggingOpts(String logDirectory, Configuration conf) {
+		boolean enableGCLogging = conf.getBoolean(CoreOptions.FLINK_JVM_DEFAULT_GC_LOGGING);
+		if (enableGCLogging) {
+			return "-Xloggc:" + logDirectory + "/gc.log " +
+				"-XX:+PrintGCApplicationStoppedTime " +
+				"-XX:+PrintGCDetails " +
+				"-XX:+PrintGCDateStamps " +
+				"-XX:+UseGCLogFileRotation " +
+				"-XX:NumberOfGCLogFiles=10 " +
+				"-XX:GCLogFileSize=10M " +
+				"-XX:+PrintPromotionFailure " +
+				"-XX:+PrintGCCause";
+		}
+		return "";
+	}
+
+	/**
+	 * Format the default heapdump options
+	 * @param appId application id
+	 * @param ident the ident of the process, taskmanager/jobmanager
+	 * @param logDirectory to print some logs
+	 * @param conf flink configuration
+	 * @return the formatted heapdump options string
+	 */
+	public static String getHeapdumpOpts(String appId, String ident, String logDirectory, Configuration conf) {
+		boolean enableHeapDump = conf.getBoolean(CoreOptions.FLINK_JVM_HEAPDUMP_ON_OOM);
+		String heapdumpDir = conf.getString(CoreOptions.FLINK_JVM_HEAPDUMP_DIRECTORY);
+		if (enableHeapDump) {
```

Comment:
Same here with the check whether to return heap dump opts or not. If this feature is not enabled, then we should not call into this method here.

Reply:
Done